### ISSUE TITLE
Not able to use CustomServlet without urlpattern /*

### ISSUE THREAD

**Issue by neerup** [vaadin/multiplatform-runtime#29]
Hi... 
I'm in the process of converting a Vaadin 8 projekt to Vaadin 11 using MPR
But I'm not able to use a Custom Vaadin servlet without a servlet mapping to /*
I have tried:
 `@WebServlet(urlPatterns = { "/ui/*", "/frontend/*", "/framework/*", "/VAADIN/*" }, name = "UIServlet", asyncSupported = true)` but i'm not able to load my custom Widgetset.
Getting 404 on : http://localhost:8080/myapp/ui/framework/VAADIN/widgetsets/AppWidgetset/AppWidgetset.nocache.js?1542231209848 net::ERR_ABORTED 404
Any help is appreciated!!!
Cheers..

**Comment by caalador** [comment]
Is the custom servlet for flow or framework?

**Comment by neerup** [comment]
Servlet is for flow....

**Comment by caalador** [comment]
At least `/framework/*` should be registered by `MprServlet` that is a Framework VaadinServlet and not by Flow.

There is a possibility that the LegacyWrapper generated configuration would need to be made more configurable as now the sent `vaadinDir`is always `./framework/VAADIN/` and the servlet url used is `./framework/`

I guess that if trying `http://localhost:8080/myapp/framework/VAADIN/widgetsets/AppWidgetset/AppWidgetset.nocache.js?1542231209848` then the widgetset would be found, but now it has `myapp/ui/` which is one level wrong. You could try having  instead of `"/framework/*"` ` "/ui/framework/*"`

**Comment by neerup** [comment]
caaldor am I missing something... will /ui/* not pickup /ui/framework/* requests ?

**Comment by caalador** [comment]
It should, but the problem is that the `/ui/framework` requests should go to the `MprServlet` that handles framework requests.

**Comment by neerup** [comment]
OK... so is there a solution or something i should try?

**Comment by caalador** [comment]
It feels like something should be changed to support this.

If you have a sample use case of what and why we could then investigate and perhaps come up with a solution for changing the servlet mapping on the Servlet.

**Comment by neerup** [comment]
This is exactly the same as issue #4619 from Flow:
All Flow apps and features should work from non-root servlet context #4619

**Comment by alump** [comment]
Why this can't be resolved by replacing the hardcoded "./framework"s on this specific class with method:

```
protected String getServletUrl() {
    return VaadinService.getCurrentRequest().getContextPath() + "/framework";
}
```

**Comment by mehdi-vaadin** [comment]
@alump was your PR (https://github.com/vaadin/multiplatform-runtime-internal/pull/339) an attempt to fix this issue?

**Comment by caalador** [comment]
@mehdi-vaadin #339 was repaced and fixed with #358
So this should be marked as closed I guess.

**Comment by mehdi-vaadin** [comment]
> @mehdi-vaadin #339 was repaced and fixed with #358
> So this should be marked as closed I guess.
I noticed that. But, I didn't see any explicit relation between this and either of those. Just wanted to make sure.
Ok, I'm closing this :)
